src/app.ts

```diff
@@ -1,24 +1,73 @@
-const Koa = require('koa');
-const cors = require('@koa/cors');
+#!/usr/bin/env node
 
-const { admin } = require('./firebaseAdmin');
-const { verifyFirebaseIdToken } = require('./verifyFirebaseIdToken');
+import path from 'path';
+
+import Koa from 'koa';
+import cors from '@koa/cors';
+import Router from 'koa-router';
+
+import { getAdmin } from './firebaseAdmin';
+import  { verifyFirebaseIdToken } from './verifyFirebaseIdToken';
+
+let admin;
+if (process.env.NODE_ENV === 'test' && process.argv.length > 2) {
+    const credentialsPathArg = process.argv[2];
+    if (path.isAbsolute(credentialsPathArg)) {
+        admin = getAdmin(credentialsPathArg);
+    } else {
+        // Resolve the path relative to the cwd.
+        admin = getAdmin(path.resolve(path.join(process.cwd(), credentialsPathArg)));
+    }
+} else {
+    admin = getAdmin();
+}
+
+const db = admin.firestore();
+const members = db.collection('members');
+const operations = db.collection('operations');
 
 const app = new Koa();
 
 app.use(cors());
 
-app.use(verifyFirebaseIdToken);
+// Put endpoints that don't need the user to be authenticated above this.
+app.use(verifyFirebaseIdToken(admin));
+// Put endpoints that do need the user to be authenticated below this.
 
-app.use(async ctx => {
-    // const db = admin.database();
-    // const members = db.collections('members');
-    ctx.body = ctx.url + ' ' + ctx.state.user; // + ' ' + members;
-});
+const authenticatedRouter = new Router()
+    .param('mid', async (mid, ctx, next) => {
+        const midQuery = (await members.where('mid', '==', mid).get());
```

Comment:
My bias would be to use uid since client already has it, but unless we have strong data showing this is slower/costlier this should be fine.